doc/generating-idyntree-matlab-bindings.md

```diff
@@ -0,0 +1,58 @@
+
+As per the documentation of [iDynTree](https://github.com/robotology/idyntree) for [modifications of Matlab/Octave bindings](https://github.com/robotology/idyntree#matlaboctave-bindings-modifications), the prerequisite is an experimental version of [Swig](http://www.swig.org/svn.html) with Matlab support.
+This can be found in the repository [https://github.com/robotology-dependencies/swig/](https://github.com/robotology-dependencies/swig/) (branch `matlab`).
+
+## Installing experimental version of Swig
+- Clone the repository from [https://github.com/robotology-dependencies/swig/](https://github.com/robotology-dependencies/swig/)
+- Install the required dependencies for SWIG listed in the **Required Tools** section of the [Swig page](http://www.swig.org/svn.html).
+- Follow the installation instructions in [http://www.swig.org/Doc3.0/SWIGDocumentation.html#Preface_unix_installation](http://www.swig.org/Doc3.0/SWIGDocumentation.html#Preface_unix_installation) to install Swig locally. By default SWIG installs itself in `/usr/local`, in Unix environment. However, it is preferable to install Swig in a different location so as to avoid conflicts with other installed versions of SWIG.
+- Proper installation of SWIG will generate a `swig` executable in `<swig_install_dir>/bin`.
+
+This executable will be used to generate the iDynTree Matlab/Octave bindings for custom classes.
+
+## Including the custom classes for bindings generations
+In this section, as a toy example, let us consider a custom class `DiscreteKalmanFilterHelper` for which the bindings need to be generated. Let's assume the class `DiscreteKalmanFilterHelper` exists in the header file `iDynTree/Estimation/KalmanFilter.h`.
+
+- Take a glance at `bindings/iDynTree.i`. It follows a particular structure, adding the necessary header files twice for the desired classes, one using the prefix `#` and another using the prefix `%`.
+    - It is also important to notice the order in which these headers are added, which ensures that the dependent header file is compiled after the dependee header file.
+    - There are also sections for adding library specific data-structure; sensors-related, estimation-related, etc.
+    - Templatized classes are explicitly specialized, so that the bindings are generated for the template specializations.
+
+- Coming back to our toy example, we will add the header file in the section related to the `Estimation related classes`, in two different places
+```
+#include "iDynTree/Estimation/KalmanFilter.h"
+```
+and
+```
+%include "iDynTree/Estimation/KalmanFilter.h"
+```
+
+Now, all is left is to compile and generate the bindings.
+
+## Generating the bindings
+- In the build folder of `iDynTree`, run `ccmake` to open the CMake Gui.
+    - Set the CMake option `SWIG_EXECUTABLE` to `<swig_install_dir>/bin/swig`.
+    - Set the CMake option `IDYNTREE_GENERATE_MATLAB` to `ON`.
+    - Set the CMake option `IDYNTREE_USES_MATLAB` to `ON`.
```

Comment:
```suggestion 
    - Set the CMake option `IDYNTREE_USES_MATLAB` or  `IDYNTREE_USES_OCTAVE`  to `ON`.
```

Reply:
Fixed in https://github.com/robotology/idyntree/commit/142bf0545b29f47c11325458657e779532bf9810